larrecodnn/CVN/modules/LArCVNEvaluator_module.cc

```diff
@@ -0,0 +1,132 @@
+// \author  Alexander Radovic - a.radovic@gmail.com
+//          Saul Alonso Monsalve - saul.alonso.monsalve@cern.ch
+////////////////////////////////////////////////////////////////////////
+
+// C/C++ includes
+#include <iostream>
+#include <sstream>
+
+// Framework includes
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art/Utilities/make_tool.h"
+#include "art_root_io/TFileDirectory.h"
+#include "art_root_io/TFileService.h"
+#include "canvas/Persistency/Common/Assns.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "larrecodnn/CVN/func/AssignLabels.h"
+#include "larrecodnn/CVN/func/InteractionType.h"
+#include "larrecodnn/CVN/func/PixelMap.h"
+#include "larrecodnn/CVN/func/Result.h"
+#include "larrecodnn/CVN/interfaces/ITFNetHandler.h"
+
+namespace lcvn {
+
+  class LArCVNEvaluator : public art::EDProducer {
+  public:
+    explicit LArCVNEvaluator(fhicl::ParameterSet const& pset);
+    ~LArCVNEvaluator();
+
+    void produce(art::Event& evt);
+    void beginJob();
+    void endJob();
+
+  private:
+    /// Module label for input pixel maps
+    std::string fPixelMapInput;
+    std::string fResultLabel;
+
+    /// Can use Caffe or Tensorflow
+    std::string fCVNType;
+
+    //lcvn::CaffeNetHandler fCaffeHandler;
+    std::unique_ptr<lcvn::ITFNetHandler> fTFHandler;
+
+    /// Number of outputs fron neural net
+    //unsigned int fNOutput;
+
+    /// If there are multiple pixel maps per event can we use them?
+    bool fMultiplePMs;
+  };
+
+  //.......................................................................
+  LArCVNEvaluator::LArCVNEvaluator(fhicl::ParameterSet const& pset)
+    : EDProducer{pset}
+    , fPixelMapInput(pset.get<std::string>("PixelMapInput"))
+    , fResultLabel(pset.get<std::string>("ResultLabel"))
+    , fCVNType(pset.get<std::string>("CVNType"))
+    ,
+    //fCaffeHandler       (pset.get<fhicl::ParameterSet> ("CaffeNetHandler")),
+    fTFHandler{art::make_tool<ITFNetHandler>(pset.get<fhicl::ParameterSet>("TFHandler"))}
+    ,
+    //fNOutput       (fCaffeHandler.NOutput()),
+    fMultiplePMs(pset.get<bool>("MultiplePMs"))
+  {
+    produces<std::vector<lcvn::Result>>(fResultLabel);
+  }
+  //......................................................................
+  LArCVNEvaluator::~LArCVNEvaluator()
+  {
+    //======================================================================
+    // Clean up any memory allocated by your module
+    //======================================================================
+  }
+
+  //......................................................................
+  void LArCVNEvaluator::beginJob() {}
+
+  //......................................................................
+  void LArCVNEvaluator::endJob() {}
```

Comment:
Suggest removing these three functions--the destructor does nothing, and because the `beginJob` and `endJob` functions are not doing anything, there's no need to provide them.